sklearn/utils/estimator_checks.py

```diff
@@ -2939,6 +2944,65 @@ def check_classifiers_regression_target(name, estimator_orig,
             e.fit(X, y)
 
 
+@ignore_warnings(category=(DeprecationWarning, FutureWarning))
+def check_estimator_sparse_dense(name, estimator_orig,
+                                 strict_mode=True):
+    rng = np.random.RandomState(42)
+    estimator = clone(estimator_orig)
+    estimator_sp = clone(estimator_orig)
+    if is_regressor(estimator_orig):
+        n_samples = 10
+        n_features = 2
+        X = sparse.random(n_samples, n_features, density=0.8,
+                          random_state=rng).A.astype(np.float16)
+        X = X.astype(np.float)
+        # Less significative numbers for less numerical pbs
+        X = X[~np.all(X == 0, axis=1)]  # we remove null rows
+        assert len(np.where(X == 0)[0]) > 0  # we should have sparsity
+        assert X.shape[0] > 5  # we mustn't have very few samples
+        size_X = X.shape[0]
+        y = X.dot(np.array([0.1, -0.2])) + 0.01 * rng.randn(size_X)
+        y = y.astype(np.float16).astype(np.float)
```

Comment:
add a note why you added this.

Reply:
done here: https://github.com/scikit-learn/scikit-learn/pull/13246/commits/70d6e889d5c1d2ba942601f7afa3f41537db0564, let me know what you think about it (I'm not completely sure of it):

```
        # we build float64 precision numbers from low precision ones:
        # this way, fewer numbers get dropped by the
        # machine precision in multiplications/additions, so if those
        # are done in a different order btw sparse/dense, the results
        # are more likely to still be the same
```